PlaceRouteHierFlow/placer/ILP_solver.cpp

```diff
@@ -272,7 +272,7 @@ double ILP_solver::GenerateValidSolution(design& mydesign, SeqPair& curr_sp, PnR
     for (int i = 0; i < mydesign.Nets.size(); i++) {
       vector<pair<int, int>> blockids;
       for (int j = 0; j < mydesign.Nets[i].connected.size(); j++) {
-        if (mydesign.Nets[i].connected[j].type == placerDB::Block)
+        if (mydesign.Nets[i].connected[j].type == placerDB::Block && (blockids.size() == 0 || mydesign.Nets[i].connected[j].iter2 != curr_sp.negPair[blockids.back().first]))
```

Comment:
If a block has 2 or more pins in one net, when adding lpsolve constraint, one variable is used more than once and the solver raise error. Now keep the first one.